Source/UZKFileInfo.m

```diff
@@ -105,4 +107,20 @@ - (NSDate *)readDate:(tm_unz)date
     return [calendar dateFromComponents:components];
 }
 
+- (BOOL)isDirectoryWith:(unz_file_info64 *)fileInfo {
+    
+    uLong type = fileInfo->external_fa >> 0x1D & 0x1F;
+    if (0 == (fileInfo->version >> 8)) {    //is DOS-archive
+        type = fileInfo->external_fa >> 4;
+        return (type == 0x01) && ![self isSymLinkWith:fileInfo->external_fa];
+    }
+    return (0x02 == type)  && ![self isSymLinkWith:fileInfo->external_fa];
+}
+
+- (BOOL)isSymLinkWith:(uLong)externalFileAttributes {
+    
+    uLong type = externalFileAttributes >> 0x1D & 0x1F;
+    return 0x05 == type;
```

Comment:
Can you send a link to the documentation for the symlinks bit? I'm not finding anything indicating where the `0x05` file means the entry is a symbolic link.

Reply:
I got more information from [this link](Https://stackoverflow.com/questions/434641/how-do-i-set-permissions-attributes-on-a-file-in-a-zip-file-using-pythons-zip), but since I am on vacation, I am temporarily unable to submit new code.

Reply:
I don't see from that link where the constants you used are.